### ISSUE TITLE
Problem with SST fields from CESM model (test_data)

### ISSUE THREAD

**Issue by MarcoLandtHayen** [MarcoLandtHayen/climate_index_collection#118]
There seems to be a problem with SST data from CESM model: There are large regions (over ocean) with SST=0. I realized that, because "zero mean" test for Western Subtropical Indian Ocean SST anomaly index failed, since mean and std dev are ZERO and dividing by ZERO leads to NaN values in resulting index, which then has no valid mean value.

How to fix this:
- don't flip CESM SSS latitude here: https://github.com/MarcoLandtHayen/climate_index_collection/blob/b06bda533bab7a8aa1ba2fe92855d71757c15f3f/data/test_data/regrid_CESM_SSS_data.py#L86
- Repeat the regridding for, both, the full and the test data SSS for CESM

Details in the comments below.

**Comment by willirath** [comment]
This seems to be a problem with the lat / lon order as
```python
from climate_index_collection.data_loading import load_data_set

ds_CESM = load_data_set(data_path="../data/test_data/", data_source_name="CESM")
ds_CESM["sea-surface-salinity"].isel(time=0).plot()
```
yields the following figure:

![image](https://user-images.githubusercontent.com/5700886/187946076-3cc79ffd-41d8-4a36-b16b-fceae9cd6b6b.png)

**Comment by willirath** [comment]
Is this also wrong in the full data?

**Comment by willirath** [comment]
It's not even wrong in the raw data:
```python
import xarray as xr

ds_raw = xr.open_dataset(
    "../data/test_data/CESM/B1850WCN_f19g16_1000y_v3.2_mod-S15-G16.cam2.h0.0001-0010.SST.nc"
)
ds_raw

ds_raw.SST.isel(time=0).plot()
```
yields
![image](https://user-images.githubusercontent.com/5700886/187947093-a43707ca-a539-4836-a881-4fc73ff52484.png)

**Comment by willirath** [comment]
So it's in the data-loading code.

**Comment by willirath** [comment]
Ignore what I've written above: It's `sea-surface-salinity` where lat is flipped. This happened here, I think:
https://github.com/MarcoLandtHayen/climate_index_collection/blob/b06bda533bab7a8aa1ba2fe92855d71757c15f3f/data/test_data/regrid_CESM_SSS_data.py#L86
And then, SST is masked using https://github.com/MarcoLandtHayen/climate_index_collection/blob/b06bda533bab7a8aa1ba2fe92855d71757c15f3f/climate_index_collection/data_loading.py#L154
which uses SSS for constructing the mask.

The resulting SST looks like this:
![image](https://user-images.githubusercontent.com/5700886/187949376-414f71e2-c159-4b55-bb4f-156522b6c0ae.png)

**Comment by MarcoLandtHayen** [comment]
Closed by #117
